app/main.py

```diff
@@ -1,3 +1,81 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __add__(self, other: int) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        else:
+            raise TypeError("Unsupported type for addition of distance")
+
+    def __iadd__(self, other: int) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        else:
+            raise TypeError("Unsupported type for addition")
+        return self
+
+    def __mul__(self, other: int) -> Distance:
```

Comment:
The type annotation for the `other` parameter should be `Union[int, float]` instead of just `int` to accurately reflect the types that can be multiplied with a `Distance` object.